src/extensions/NEventExtension.ts

```diff
@@ -72,17 +67,12 @@ export const NEventExtension = Node.create({
   addCommands() {
     return {
       insertNEvent:
-        ({ nevent }) =>
-        ({ commands }) => {
-          const parts = nevent.split(':')
-          const attrs = nip19.decode(parts[parts.length - 1])?.data as EventPointer
-          return commands.insertContent(
-            { type: this.name, attrs: { ...attrs, nevent } },
-            {
-              updateSelection: false,
-            },
-          )
-        },
+        ({ entity }) =>
```

Comment:
I feel like the nevent/entity needs some sort of validation at the extension level as we don't want users to pass a nprofile here, and the `entityToPointer` will return NProfilePointer anyway

Reply:
I agree, I've added some validation via options